app/src/main/java/com/quran/labs/androidquran/util/QuranFileUtils.java

```diff
@@ -355,8 +355,21 @@ public static String getQuranAyahDatabaseDirectory(Context context) {
 
   @Nullable
   public static String getQuranAudioDirectory(Context context){
-    String s = QuranFileUtils.getQuranBaseDirectory(context);
-    return (s == null)? null : s + AUDIO_DIRECTORY + File.separator;
+    String path = getQuranBaseDirectory(context);
+    if (path == null) {
+      return null;
+    }
+    path += AUDIO_DIRECTORY;
+    File dir = new File(path);
+    if (dir.exists() && !dir.isDirectory() && !dir.delete()) {
```

Comment:
This handles the case when there is an file (not a directory) with name `audio` under `quran_android`. Very unlikely case. If you think that is an overkill, please let me know and I will remove this block.